src/jvmTest/kotlin/fr/acinq/lightning/Node.kt

```diff
@@ -16,7 +16,7 @@ import fr.acinq.lightning.db.OutgoingPayment
 import fr.acinq.lightning.db.sqlite.SqliteChannelsDb
 import fr.acinq.lightning.io.*
 import fr.acinq.lightning.payment.PaymentRequest
-import fr.acinq.lightning.serialization.v1.Serialization
+import fr.acinq.lightning.serialization.v3.Serialization
```

Comment:
Why are we still using v3 serialization for anything here?

Reply:
I forgot 🤷‍♂️ ... but this is why those should be private in theory.

Given the half-finished state of this `Node`, I'm really tempted to nuke it for now, until we redo it better. WDYT? (cc @sstone)

Reply:
We could nuke it but it's still very useful for testing, and it just needs to be able to serialize channel states to JSON so that should still be possible with the changes in this PR ?

Reply:
> it's still very useful for testing

I think this is much less true since we have automated integration testing, but I see your point.

> it just needs to be able to serialize channel states to JSON so that should still be possible with the changes in this PR ?

Yes it's possible, but it forces me to expose the `Json` object, which is a bit footgunny for callers because using external serializers cause way more runtime errors depending on how it is called. That's why I defined a well-tested `fun toJsonString(state: ChannelState): String` function that ensure the correct type is used. I'll add a warning and see how it looks.

Reply:
Or you could just expose a `toJsonElement()` ?

Reply:
I updated the `Node` in eede7fd4, but as expected the `GET /channels` method fails because it can't find a serializer for `ChannelState`.

Reply:
Phew, upped my json game a bit and got it to work in e6ae89b0. Unhandled states are serialized like so:
```
{
    "type": "fr.acinq.lightning.json.JsonSerializers.ChannelStateSurrogateSerializer",
    "forState": "fr.acinq.lightning.channel.WaitForOpenChannel"
}
```

Note that calling `JsonSerializers.json.encodeToString(state: Normal)` will fail due to [static parent type lookup](https://github.com/Kotlin/kotlinx.serialization/blob/master/docs/polymorphism.md#static-parent-type-lookup-for-polymorphism). This is what I was initially afraid of, but I think it's fair game, because we always want the `type` attribute. Any of `JsonSerializers.json.encodeToString(state: ChannelState)` or `JsonSerializers.json.encodeToString(state: ChannelStateWithCommitments)` or `JsonSerializers.json.encodeToString(state: PersistedChannelState)` work now.

⚠️ Just noticed that `Offline` and `Syncing` are `ChannelState`, not `ChannelStateWithCommitments`. That's another design flaw, due to the fact that those states are not persisted. But the confusion has been cleared up with `PersistedChannelState`. I'm afraid of changing the type though, who knows what it can break.